src/sage/matroids/circuits_matroid.pyx

```diff
@@ -492,103 +467,105 @@ cdef class CircuitsMatroid(Matroid):
              frozenset({2, 3})]
         """
         from itertools import combinations
-        cdef set B = set()
         cdef set NB = set(self.nonbases())
         cdef frozenset S
-        for SS in combinations(self._groundset, self._matroid_rank):
-            S = frozenset(SS)
+        for St in combinations(self._groundset, self._matroid_rank):
+            S = frozenset(St)
             if S not in NB:
                 yield S
 
-    cpdef nonbases(self):
-        r"""
-        Return the nonbases of the matroid.
-
-        OUTPUT: :class:`SetSystem`
-
-        EXAMPLES::
-
-            sage: from sage.matroids.circuits_matroid import CircuitsMatroid
-            sage: M = CircuitsMatroid(matroids.Uniform(2, 4))
-            sage: len(M.nonbases())
-            0
-            sage: M = CircuitsMatroid(matroids.CompleteGraphic(6))
-            sage: len(M.nonbases())
-            1707
-        """
-        return self.dependent_r_sets(self._matroid_rank)
-
-    cpdef independent_r_sets(self, long r):
+    cpdef SetSystem independent_sets(self, long k=-1):
         r"""
-        Return the size-``r`` independent subsets of the matroid.
+        Return the size-`k` independent sets of the matroid.
 
         INPUT:
 
-        - ``r`` -- nonnegative integer
+        - ``k`` -- integer
```

Comment:
Shouldn't this explain what happens when `k` is not provided

Reply:
Yes, thanks.